File: src/main/java/com/baulsupp/oksocial/authenticator/basic/BasicAuthServiceDefinition.java
package com.baulsupp.oksocial.authenticator.basic;

import com.baulsupp.oksocial.AbstractServiceDefinition;
import com.baulsupp.oksocial.authenticator.BasicCredentials;

public class BasicAuthServiceDefinition extends AbstractServiceDefinition<BasicCredentials> {
  public BasicAuthServiceDefinition(String apiHost, String serviceName, String shortName,
      String apiDocs, String accountsLink) {
    super(apiHost, serviceName, shortName, apiDocs, accountsLink);
  }

  public BasicCredentials parseCredentialsString(String s) {
    String[] parts = s.split(":", 2);
    return new BasicCredentials(parts[0], parts[1]);
  }

  public String formatCredentialsString(BasicCredentials credentials) {
    return credentials.user + ":" + credentials.password;
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/authenticator/basic/BasicAuthServiceDefinition.kt
package com.baulsupp.oksocial.authenticator.basic

import com.baulsupp.oksocial.AbstractServiceDefinition
import com.baulsupp.oksocial.authenticator.BasicCredentials

class BasicAuthServiceDefinition(apiHost: String, serviceName: String, shortName: String,
                                 apiDocs: String, accountsLink: String?) : AbstractServiceDefinition<BasicCredentials>(apiHost, serviceName, shortName, apiDocs, accountsLink) {

  override fun parseCredentialsString(s: String): BasicCredentials {
    val parts = s.split(":".toRegex(), 2).toTypedArray()
    return BasicCredentials(parts[0], parts[1])
  }

  override fun formatCredentialsString(credentials: BasicCredentials) =
      credentials.user + ":" + credentials.password
}
